src/txdb.cpp

```diff
@@ -103,7 +103,12 @@ bool CCoinsViewDB::GetStats(CCoinsStats &stats) const {
     pcursor->Seek(DB_COINS);
 
     CHashWriter ss(SER_GETHASH, PROTOCOL_VERSION);
-    stats.hashBlock = GetBestBlock();
+    
+    {
+      LOCK(cs_main);
+      stats.hashBlock = GetBestBlock();
+      stats.nHeight = mapBlockIndex.find(stats.hashBlock)->second->nHeight;
```

Comment:
I was concerned that a chain re-org occurring in another thread might make mapBlockIndex.find return end() if stats.hashBlock is no longer on the main chain.